src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -286,6 +290,25 @@ func (c *AggregatorConfiguration) NewAggregatorOptions(
 	}
 	opts = opts.SetFlushHandler(flushHandler)
 
+	// Set pass-through handler.
+	passThroughScope := scope.SubScope("pass-through-handler")
+	iOpts = instrumentOpts.SetMetricsScope(passThroughScope)
+	// fishie9: this is a temporary change to use a separate m3msg topic for pass-through metrics during the migration
+	for _, handler := range c.Flush.Handlers {
+		if handler.DynamicBackend != nil && c.PassThroughTopicName != nil {
+			handler.DynamicBackend.Producer.Writer.TopicName = *c.PassThroughTopicName
+		}
+	}
+	passThroughHandler, err := c.Flush.NewHandler(client, iOpts)
+	if err != nil {
+		return nil, err
+	}
+	passThroughWriter, err := passThroughHandler.NewWriter(passThroughScope)
+	if err != nil {
+		return nil, err
+	}
+	opts = opts.SetPassThroughWriter(writer.NewMutexWriter(passThroughWriter))
```

Comment:
i'm probably missing something here, but what's the point of wrapping the `passThroughWriter` with a `writer.NewMutexWriter(...)`?